src/ColumnMetrics.js

```diff
@@ -27,60 +27,68 @@ type Column = {
  *
  * @param {ColumnMetricsType} metrics
  */
-function calculate(metrics: ColumnMetricsType): ColumnMetricsType {
-  var width = 0;
-  var unallocatedWidth = metrics.totalWidth;
-
-  var deferredColumns = [];
-  var columns = metrics.columns.map(shallowCloneObject);
-
-  var i, len, column;
+function recalculate(metrics: ColumnMetricsType): ColumnMetricsType {
```

Comment:
reactDataGrid has no concept of hidden columns. Another one for the issues list methinks

https://github.com/adazzle/react-data-grid/issues/27